hazelcast/src/test/java/com/hazelcast/jet/impl/connector/WriteJdbcPTest.java

```diff
@@ -101,19 +102,24 @@ public static void setupClass() throws SQLException {
 
     @Before
     public void setup() throws SQLException {
-        tableName = "T" + tableCounter.incrementAndGet();
+        tableName = "T" + TABLE_COUNTER.incrementAndGet();
         logger.info("Will use table: " + tableName);
-        try (Connection connection = ((DataSource) createDataSource(false)).getConnection()) {
-            connection.createStatement()
-                    .execute("CREATE TABLE " + tableName + "(id int, name varchar(255))");
-        }
+        executeSql("CREATE TABLE " + tableName + "(id int, name varchar(255))");
     }
 
     @After
     public void tearDown() throws Exception {
         if (hikariDataSource != null) {
             hikariDataSource.close();
         }
+        // kill any hanging connection
```

Comment:
Do you know why we have hanging connections? 

I wonder if it's a resource leaks related to `JdbcDataStoreFactory` or close to its usage

Could list the hanging connections? e.g. with `SELECT * FROM pg_stat_activity WHERE pid <> pg_backend_pid()` ?.

Reply:
I couldn't reproduce the issue, but my guess is that we are making forceful job restart in some cases, so `close` may be not called if forceful restart was in wrong moment